popx/migration_box.go

```diff
@@ -146,3 +199,24 @@ func (fm *MigrationBox) findMigrations(runner func([]byte) func(mf Migration, c
 		return nil
 	})
 }
+
+// hasDownMigrationWithVersion checks if there is a migration with the given
+// version.
+func (fm *MigrationBox) hasDownMigrationWithVersion(version string) bool {
+	for _, down := range fm.Migrations["down"] {
+		if version == down.Version {
+			return true
+		}
+	}
+	return false
+}
+
+// check checks that every "up" migration has a corresponding "down" migration.
+func (fm *MigrationBox) check() error {
+	for _, up := range fm.Migrations["up"] {
+		if !fm.hasDownMigrationWithVersion(up.Version) {
+			return fmt.Errorf("migration %s has no corresponding down migration", up.Version)
```

Comment:
```suggestion
			return errors.Errorf("migration %s has no corresponding down migration", up.Version)
```
Assuming the `github.com/pkg/errors` package. This error will automatically have a stack trace, which we inside of Ory always want to have.